### ISSUE TITLE
Concept: TriProtect

### ISSUE THREAD

**Issue by ethamitc** [Triangle-Project/Triangle#4]
This is a system that will protect Triangle itself from malicious users and guilds by using detection systems combined with human intervention to determine whether a user/guild is malicious in intent.

Any guild deemed (backed by human confirmation) to be malicious will be automatically removed by the bot and any attempts to re-add the bot will result in the same.

Any user deemed (backed by human confirmation) to be malicious will be added to a database where any server they join the admins will be notified, and if the server is opted into the TriProtect global ban list, they will be banned automatically.

Obviously since these systems inhibit the use of Triangle itself (server level), and server access itself (user level, if enabled), **all** actions will performed by the Triangle staff team, with automated systems in place for **detection only**. This will ensure that the system will can't mistakenly ban a guild or user as this would result in people being upset.

As noted in the title, this is a concept and is open to changes/additions, however I feel this system should 100% be implemented as it will help keep abuse of Triangle down (server level), as well as help keep servers free of malicious users.

**Comment by Kiddo9000** [comment]
I think this sounds good on paper, but it could have unwanted negative attention. Having the bot forcefully remove itself from a server could generate negative backlash from people and cause people to attempt to abuse it more on purpose. This can also be avoided by making a new server since if people really want to use it for malicious activities they will easily move servers to continue doing so.

However, I do have some ideas that we could use instead. One for sure is to remove invites to known malicious servers and warn users not to join the server. This could help prevent the spread of the malicious server and damage its ability to grow and gain more members. Another would be to ignore statistical results and disable malicious user warnings on those guilds, basically treating the entire server's members as potentially malicious. Infractions caused by users should also not be logged/added to the database OR should be flagged to be ignored, since malicious guilds are far more likely to warn and/or ban innocent users, thus avoiding damaging an innocent users infraction count.

TL;DR: The idea is great but having the bot remove itself could cause unwanted negative rep and seems a little too much. Instead, it should do the following for malicious guilds:
- Remove invites to malicious servers and warn users that the server might be harmful.
- Disable both SafeServer and TriProtect and mark the server as 'ineligible' for the features. This means no warnings for potentially malicious users who join and no auto bans/kicks. Admins should also not be able to check a user's rating manually.
- Hide the server from all statistics and listings.
- Flag/ignore all warns, kicks, and bans in the database from malicious servers. The server is more likely to kick/ban an innocent user, which can affect their infraction count when most of the time, they did nothing wrong.

**Comment by therealastrid** [comment]
I like the idea of protecting it from bad guilds, but i think that judging them is bad. Also against the TOS to collect that data.

**Comment by ethamitc** [comment]
> TL;DR: The idea is great but having the bot remove itself could cause unwanted negative rep and seems a little too much. Instead, it should do the following for malicious guilds:
> 
> * Remove invites to malicious servers and warn users that the server might be harmful.
> * Disable both SafeServer and TriProtect and mark the server as 'ineligible' for the features. This means no warnings for potentially malicious users who join and no auto bans/kicks. Admins should also not be able to check a user's rating manually.
> * Hide the server from all statistics and listings.
> * Flag/ignore all warns, kicks, and bans in the database from malicious servers. The server is more likely to kick/ban an innocent user, which can affect their infraction count when most of the time, they did nothing wrong.

I think the idea of marking a server as `ineligable` for protection features is a better idea then straight up removing. However, I do still feel that the Triangle team *should* be able to "ban" a guild because just as having it banned from a server can give it a bad rep, so can people joining a malicious server and seeing Triangle as a bot in said server.



> I like the idea of protecting it from bad guilds, but i think that judging them is bad. Also against the TOS to collect that data.

Not against TOS because everything is on a user-report basis.
